### PR TITLE
enhancement: Add support to open review page from custom test webview

### PR DESCRIPTION
**Pull request by PruthiviRaj27**
- In this commit, we introduced a JavaScript interface `showReview` in CustomTestGenerationViewController to display the review page on the IOS App. Now, when users click on a review, we fetch the corresponding attempt using the `attemptId` and open the review page.

### CODE REVIEW

File: ios-app/UI/TestReportViewController.swift [hunk 1]
```diff
@@ -251,6 +251,10 @@ class TestReportViewController: UIViewController {
             presentingViewController?.presentingViewController as? ContentDetailPageViewController {
             
             goToContentDetailPageViewController(contentDetailPageViewController)
+        } else if let customTestViewController =
+                    self.presentingViewController as? CustomTestGenerationViewController {
+            
+            customTestViewController.presentingViewController?.dismiss(animated: true, completion: nil)
```
Comment (harinath01) [inline ios-app/UI/TestReportViewController.swift:257]: This part is not clear, what are you doing? if possible extract it into function with the intention revealing name.

File: ios-app/UI/CustomTestGenerationViewController.swift [hunk 2]
```diff
@@ -45,6 +46,12 @@ class CustomTestGenerationViewController: WebViewController, WKScriptMessageHand
             print(attemptId)
             loadAttempts(attemptId as! String, true)
         }
+        if message.name == "showReview" {
+            self.emptyView.hide()
+            self.activityIndicator?.startAnimating()
+            let attemptId = message.body
+            getAttempt(attemptId as! String)
```
Comment (Karthik-0) [inline ios-app/UI/CustomTestGenerationViewController.swift:53]: Extract

File: ios-app/UI/CustomTestGenerationViewController.swift [hunk 3]
```diff
@@ -45,6 +46,16 @@ class CustomTestGenerationViewController: WebViewController, WKScriptMessageHand
...
+        if message.name == "showReview" {
+            handleShowReviewMessage(message)
+        }
+    }
+    
+    func handleShowReviewMessage(_ message: WKScriptMessage) {
+        self.emptyView.hide()
+        self.activityIndicator?.startAnimating()
+        let attemptId = message.body
+        getAttempt(attemptId as! String)
```
Comment (Karthik-0) [inline ios-app/UI/CustomTestGenerationViewController.swift:58]: What is getAttempt?

File: ios-app/UI/CustomTestGenerationViewController.swift [hunk 4]
```diff
@@ -45,6 +46,16 @@ class CustomTestGenerationViewController: WebViewController, WKScriptMessageHand
             print(attemptId)
             loadAttempts(attemptId as! String, true)
         }
+        if message.name == "showReview" {
+            handleShowReviewMessage(message)
```
Comment (Karthik-0) [inline ios-app/UI/CustomTestGenerationViewController.swift:50]: Not sure what is being handled. Function name should explain the intent.

File: ios-app/UI/CustomTestGenerationViewController.swift [hunk 5]
```diff
@@ -45,6 +46,16 @@ class CustomTestGenerationViewController: WebViewController, WKScriptMessageHand
...
+        }
+    }
+    
+    func handleShowReviewMessage(_ message: WKScriptMessage) {
+        self.emptyView.hide()
+        self.activityIndicator?.startAnimating()
+        let attemptId = message.body
+        fetchAttempt(attemptId as! String)
```
Comment (Karthik-0) [inline ios-app/UI/CustomTestGenerationViewController.swift:58]: .

File: ios-app/UI/CustomTestGenerationViewController.swift [hunk 6]
```diff
@@ -94,6 +126,14 @@ class CustomTestGenerationViewController: WebViewController, WKScriptMessageHand
         present(viewController, animated: true, completion: nil)
     }
     
+    func gotoTestReport(_ attempt: Attempt) {
+        let storyboard = UIStoryboard(name: Constants.EXAM_REVIEW_STORYBOARD, bundle: nil)
+        let viewController = storyboard.instantiateViewController(withIdentifier:
+                Constants.TEST_REPORT_VIEW_CONTROLLER) as! TestReportViewController
```
Comment (Karthik-0) [inline ios-app/UI/CustomTestGenerationViewController.swift:132]: This ViewController name itself is confusing(not sure if it means exam or test case). Would prefer it to be renamed in separate commit.

File: ios-app/UI/TestReportViewController.swift [hunk 7]
```diff
@@ -251,13 +251,21 @@ class TestReportViewController: UIViewController {
             presentingViewController?.presentingViewController as? ContentDetailPageViewController {
             
             goToContentDetailPageViewController(contentDetailPageViewController)
+        } else if let customTestViewController =
+                    self.presentingViewController as? CustomTestGenerationViewController {
+            
+            goToCourseListView(customTestViewController)
         } else if exam == nil {
```
Comment (Karthik-0) [inline ios-app/UI/TestReportViewController.swift:258]: This whole back button logic can be extracted.
It should follow this structure.
if (isPresentedFromView()) {    // Check better name from chatgpt
    goToSomeView()
}

**Review by Karthik-0 — DISMISSED** [review]

**Review by Karthik-0 — APPROVED** [review]
